common/persistence/visibility/store/elasticsearch/visibility_store.go

```diff
@@ -648,11 +651,24 @@ func (s *visibilityStore) buildSearchParametersV2(
 	request *manager.ListWorkflowExecutionsRequestV2,
 ) (*client.SearchParameters, error) {
 
-	boolQuery, fieldSorts, err := s.convertQuery(request.Namespace, request.NamespaceID, request.Query)
+	boolQuery, fieldSorts, err := s.convertQuery(
+		request.Namespace,
+		request.NamespaceID,
+		request.Query,
+	)
 	if err != nil {
 		return nil, err
 	}
 
+	// TODO(rodrigozhou): investigate possible solutions to slow ORDER BY.
+	// ORDER BY clause can be slow if there is a large number of documents and
+	// using a field that was not indexed by ES. Since slow queries can block
+	// writes for unreasonably long, this option forbids the usage of ORDER BY
+	// clause to prevent slow down issues.
+	if s.disableOrderByClause() && len(fieldSorts) > 0 {
+		return nil, query.NewConverterError("ORDER BY clause is disabled")
```

Comment:
This is not a converter, thus this shouldn't be `ConverterError`. This is more like:
```suggestion
		return nil, serviceerror.NewInvalidArgument("ORDER BY clause is not supported")
```

Reply:
I thought about `ConverterError` because it's part of converting the query to ES query. But no strong feeling about it.